### PR TITLE
buildcop function

### PR DESCRIPTION
**Pull request by dinagraves**
Python buildcop function.  Triggers off of Pub/Sub cloud-builds topic.  Parses the results and creates a junit_xml file.  Sends build results to buildcop.  The buildcop binary is included as a file in the function and is called via the subprocess routine.

### CODE REVIEW

File: testing/buildcop-function/main.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        if (status in("SUCCESS", "FAILURE") 
+            # buildcop should not run on Pull Requests
+            and "_PR_NUMBER" not in build_event["substitutions"]):
+
+            write_xml_file(build_event)
+            
+            commit_sha = build_event["substitutions"]["COMMIT_SHA"]
+            os.environ["KOKORO_BUILD_ID"] = build_event["id"]
+            out = subprocess.run(
+                ["./buildcop", "-repo=GoogleCloudPlatform/cloud-run-samples",
```
Comment (grayside) [inline testing/buildcop-function/main.py:94]: I'm mixed about having a snapshot of an 18mb binary in the repo.
We also lack a deploy script. Maybe we should add a deploy script which includes downloading the binary?

File: testing/buildcop-function/main.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# limitations under the License.
+
+import base64
+import json
+import os
+import subprocess
+from datetime import datetime, timedelta
+
+
+def write_xml_file(build_event):
```
Comment (grayside) [inline testing/buildcop-function/main.py:22]: In reviewing this, I gave some thought to the lack of test coverage and a CI pipeline on this function. I think it's okay that we skip for now, but if we increase the complexity of this function later we should think about adding tests then.

File: testing/buildcop-function/go.mod [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module github.com/GoogleCloudPlatform/cloud-run-samples/testing/buildcop-function
+
+go 1.14
+
+require github.com/googleapis/repo-automation-bots/packages/buildcop v0.0.0-20200716174820-aef5c4576904 // indirect
```
Comment (dinagraves) [inline testing/buildcop-function/go.mod:5]: Should this file be in the directory?

File: testing/buildcop-function/deploy_function.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+cp $(go env GOPATH)/bin/linux_amd64/buildcop buildcop
+
+# Deploy function
+gcloud functions deploy buildcop-worker --region=us-central1 \
+--trigger-topic=cloud-builds --runtime=python37 --entry-point=send_to_buildcop \
+--no-allow-unauthenticated
+
+# Remove binary from current directory 
+rm buildcop
+rm go.sum
```
Comment (dinagraves) [inline testing/buildcop-function/deploy_function.sh:30]: Should go.sum be in the directory? Does it make sense to remove it after the buildcop is downloaded?

File: testing/buildcop-function/main.py (hunk 1 above)
Comment (dinagraves) [reply testing/buildcop-function/main.py:94]: Removed buildcop binary and added a deploy script.

File: testing/buildcop-function/go.mod (hunk 3 above)
Comment (grayside) [reply testing/buildcop-function/go.mod:5]: The answer to this question and the one about go.sum in the script is the same:

Are we trying to pin a specific version of buildcop for download, or do we want to retrieve the latest version on each deployment? I'd argue we should "pin", in which case we should commit the go.mod and go.sum as part of managing this application.

Comment (dinagraves) [reply testing/buildcop-function/go.mod]: Added the go.sum!

**Comment by dinagraves** [comment]
Added coverage for other failure types:  https://github.com/GoogleCloudPlatform/cloud-run-samples/issues/46

**Pull request merged**